base/tks/src/main/java/org/dogtagpki/server/tks/servlet/SecureChannelProtocol.java

```diff
@@ -1003,9 +1010,19 @@ public SymmetricKey unwrapSymKeyOnToken(CryptoToken token, SymmetricKey unwrappi
         }
 
         try {
-            encryptor = token.getCipherContext(EncryptionAlgorithm.DES3_ECB);
+            EncryptionAlgorithm  encAlg = EncryptionAlgorithm.DES3_ECB;
+            KeyWrapAlgorithm wrapAlg = KeyWrapAlgorithm.DES3_ECB;
+
+            IVParameterSpec iv = null;
+            if(encUnwrapKey.getType() == SymmetricKey.Type.AES) {
+                encAlg = EncryptionAlgorithm.AES_128_CBC_PAD;
+                wrapAlg = KeyWrapAlgorithm.AES_CBC_PAD;
+                iv = new IVParameterSpec(new byte[encAlg.getIVLength()]);
+            }
+
+            encryptor = token.getCipherContext(encAlg);
 
-            encryptor.initEncrypt(encUnwrapKey);
+            encryptor.initEncrypt(encUnwrapKey,iv);
```

Comment:
Just checking...so in case iv is null (encUnwrapKey type is not AES), the initEncrypt call is going to be fine right?

Reply:
Yes, the original version of this call from the code encryptor.initEncrypt(encUnwrapKey), ends up calling initEncrypt(key,null) in the jss code. For the initUnwrap below there is code in jss that check to see if the parameter param is correct for the algorithm.